scipy/stats/_stats_py.py

```diff
@@ -1817,20 +1817,115 @@ def jarque_bera(x, *, axis=None):
+           for normality (complete samples). Biometrika, 52(3/4), 591-611.
+    .. [3] B. Phipson and G. K. Smyth. "Permutation P-values Should Never Be
+           Zero: Calculating Exact P-values When Permutations Are Randomly
+           Drawn." Statistical Applications in Genetics and Molecular Biology
+           9.1 (2010).
+    .. [4] Panagiotakos, D. B. (2008). The value of p-value in biomedical
+           research. The open cardiovascular medicine journal, 2, 97.
 
     Examples
     --------
+    Suppose we wish to infer from measurements whether the weights of adult
+    human males in a medical study are not normally distributed [2]_.
+    The weights (lbs) are recorded in the array ``x`` below.
+
     >>> import numpy as np
+    >>> x = np.array([148, 154, 158, 160, 161, 162, 166, 170, 182, 195, 236])
+
+    The Jarque-Bera test begins by computing a statistic based on the sample
+    skewness and kurtosis.
+
     >>> from scipy import stats
-    >>> rng = np.random.default_rng()
-    >>> x = rng.normal(0, 1, 100000)
-    >>> jarque_bera_test = stats.jarque_bera(x)
-    >>> jarque_bera_test
-    Jarque_beraResult(statistic=3.3415184718131554, pvalue=0.18810419594996775)
-    >>> jarque_bera_test.statistic
-    3.3415184718131554
-    >>> jarque_bera_test.pvalue
-    0.18810419594996775
+    >>> res = stats.jarque_bera(x)
+    >>> res.statistic
+    6.982848237344646
+
+    Because the normal distribution has zero skewness and zero
+    ("excess" or "Fisher") kurtosis, the value of this statistic tends to be
+    low for samples drawn from a normal distribution.
+
+    The test is performed by comparing the observed value of the statistic
+    against the null distribution: the distribution of statistic values derived
+    under the null hypothesis that the weights were drawn from a normal
+    distribution.
+    For the Jarque-Bera test, the null distribution for very large samples is
+    the chi-squared distribution with two degrees of freedom.
+
+    >>> import matplotlib.pyplot as plt
+    >>> dist = stats.chi2(df=2)
+    >>> jb_val = np.linspace(0, 11, 100)
+    >>> pdf = dist.pdf(jb_val)
+    >>> fig, ax = plt.subplots(figsize=(8, 5))
+    >>> def jb_plot(ax):  # we'll re-use this
+    ...     ax.plot(jb_val, pdf)
+    ...     ax.set_title("Jarque-Bera Null Distribution")
+    ...     ax.set_xlabel("statistic")
+    ...     ax.set_ylabel("probability density")
+    >>> ax = jb_plot(ax)
+    >>> plt.show()
+
+    The comparison is quantified by the p-value: the proportion of values in
+    the null distribution greater than or equal to the observed value of the
+    statistic.
+
+    >>> fig, ax = plt.subplots(figsize=(8, 5))
+    >>> jb_plot(ax)
+    >>> pvalue = dist.sf(res.statistic)
+    >>> annotation = (f'p-value={pvalue:.6f}\n(shaded area)')
+    >>> arrowprops=dict(facecolor='black', width=1, headwidth=5, headlength=8)
+    >>> ax.annotate(annotation, (7.5, 0.01), (8, 0.05), arrowprops=arrowprops)
```

Comment:
Let's keep the missing spaces in `arrowprops=dict` to not go overline.
And this should fix the ref guide check.
```suggestion
    >>> annotation = (f'p-value={pvalue:.6f}\n(shaded area)')
    >>> arrowprops=dict(facecolor='black', width=1, headwidth=5, headlength=8)
    >>> _ = ax.annotate(
    ...     annotation, (7.5, 0.01), (8, 0.05), arrowprops=arrowprops
    ... )
```